lib/utils/types.ts

```diff
@@ -14,22 +14,15 @@ export type Tagged<T, S, A> = t.Type<S, A & T>;
 export const tag = <T>() => <S, A>(type: t.Type<S, A>): Tagged<T, S, A> =>
   type as any;
 
-const getObjectValues = <T extends object>(obj: T): ReadonlyArray<string> =>
-  Object.keys(obj).reduce<ReadonlyArray<string>>(
-    (acc, key) => [...acc, (obj as any)[key]],
-    []
-  );
-
 /**
  * Creates an io-ts Type from an enum
  */
 export const enumType = <E>(e: {}, name: string): t.Type<any, E> => {
-  const values = getObjectValues(e);
-
   return new t.Type<any, E>(
     name,
-    (v): v is E => typeof v === "string" && values.indexOf(v) >= 0,
-    (s, c) => (this.is(s) ? t.success(s) : t.failure(s, c)),
```

Comment:
"this.is is not a function"

Reply:
ops yes, `this` is not accessible in this scope, I guess we could as well extract the `is` logic to a function:

```ts
const isE: (v: any) => boolean = v =>
    typeof v === "string" && values.indexOf(v) >= 0;

return new t.Type<any, E>(
  name,
  (v: any): v is E => isE(v),
  (s, c) => (isE(s) ? t.success(s) : t.failure(s, c)),
  t.identity
);
```

Reply:
done